### ISSUE TITLE
[Feature]: Prompt Engineering Documentation and Dev Workflow.

### ISSUE THREAD

**Issue by camigira** [k8sgpt-ai/k8sgpt#421]
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [K8sGPT Slack](https://k8sgpt.slack.com) and got positive feedback

### Is this feature request related to a problem?

None

### Problem Description

It would be good to provide some guidance around different principles, tactics and terminology around Prompt Engineering for developers. Also, a Dev Workflow to define and refine prompts.

### Solution Description

Prompt Engineering docco:
- Clear and Specific Instructions in Prompt
  - use delimiters to prevent prompt injection.
  - specify output format: JSON, YAML and define properties the model will infer.
- Give time to the model to process by specifying tasks in sequence.
- Avoid Hallucination
- Leverage summarizing and expanding capabilities.
- Leverage inferring capability.
- Leverage transformation capability.

Some guidelines/techniques when working and refining prompts, I was thinking to have a collection of broken Kubernetes objects that will cause issues. This could be used in the docs for tutorials but also for prompt engineering refinement.

### Benefits

- Increase knowledge of how LLM works and how to create and refine prompts.
- Have a framework/approach to prompt engineering
- Improve results provided by k8sgpt as LLM's evolve

### Potential Drawbacks

this might not be really a k8sgpt thing to do but perhaps a blog entry or something we can reference... perhaps there's an open-source framework in place? keen to know what people think anyway...

**Comment by camigira** [comment]
@AlexsJones thanks for your reply!

I'm not an expert myself, but I've got a real interest in delving deeper into Large Language Model's tech. Based on a few folks I follow, this awesome [course](https://www.deeplearning.ai/short-courses/chatgpt-prompt-engineering-for-developers/), and my own experimentation, I've come up with a bunch of tactics that really have helped me when it comes to prompt engineering.

So, a few weeks back, I joined the Thoughtworks [Tech Radar](https://www.thoughtworks.com/en-au/radar/techniques?blipid=202304094) webinar, and one of the creators said that Prompt Engineering isn't exactly "real" engineering, but more of a craft. Well, respectfully, I totally disagree with that! I like the following definition _"Engineering is the application of an empirical scientific approach to find efficient solutions to practical problems"_ and this can be also applied to Prompt Engineering.

I'm super keen to connect and dive deeper into this topic. Let me know when you're available. I know time zones can be a hassle, but I'm really stoked to get the ball rolling on this.
